cpp/src/io/functions.cpp

```diff
@@ -337,6 +337,35 @@ parsed_orc_statistics read_parsed_orc_statistics(source_info const& src_info)
 
   return result;
 }
+namespace {
+orc_column_schema make_orc_column_schema(host_span<orc::SchemaType const> orc_schema,
+                                         uint32_t column_id,
+                                         std::string column_name)
+{
+  std::vector<orc_column_schema> children;
+  auto const& orc_col_schema = orc_schema[column_id];
+  for (auto i = 0ul; i < orc_col_schema.subtypes.size(); ++i) {
+    children.emplace_back(make_orc_column_schema(
+      orc_schema,
+      orc_col_schema.subtypes[i],
+      i < orc_col_schema.fieldNames.size() ? orc_col_schema.fieldNames[i] : ""));
+  }
```

Comment:
Could be `std::transform` or `std::copy` for extra cudf-ness.

Reply:
There is a `std::transform` overload that takes two input iterators. My concern is the optional-ness of `fieldNames[]`. An iterator that returns a constant when it goes past the end would allow clean `std::transform` use here.

Reply:
Changed to transform. Not sure if it was worth the effort, but I learned something at least :)